reconcile/terraform_tgw_attachments.py

```diff
@@ -162,13 +218,52 @@ def run(
     if print_to_file:
         sys.exit()
 
-    aws_api = AWSApi(1, accounts, settings=settings, init_users=False)
+    return working_dirs
+
+
+@defer
+def run(
+    dry_run: bool,
+    print_to_file: Optional[str] = None,
+    enable_deletion: bool = False,
+    thread_pool_size: int = 10,
+    defer: Optional[Callable] = None,
+):
+    settings = queries.get_secret_reader_settings()
+    clusters = queries.get_clusters_with_peering_settings()
+    ocm_map = _build_ocm_map(clusters, settings)
+    accounts = queries.get_aws_accounts(terraform_state=True, ecrs=False)
+
+    # Fetch desired state for cluster-to-vpc(account) VPCs
+    with AWSApi(1, accounts, settings=settings, init_users=False) as awsapi:
+        desired_state, err = build_desired_state_tgw_attachments(
+            clusters, ocm_map, awsapi
+        )
+    if err:
+        sys.exit(1)
```

Comment:
Lets raise an exception instead - we used to do `sys.exit`, but recently agreed that we will go forward with exceptions.

Reply:
@fishi0x01 which exception we are using to replace `sys.exit(1)`?

Reply:
Good question. There is no definite standard for this. Some integrations define their own exception, some use exceptions from `reconcile.utils.exceptions`, some just `raise RuntimeError()`. Guess it depends on how descriptive you want the error to be. But as long as sth is raised its fine I guess :+1:

Reply:
updated